sbom-utility-scripts/scripts/merge-cachi2-sboms-script/merge_cachi2_sboms.py

```diff
@@ -286,14 +287,49 @@ def merge_cyclonedx_sboms(
     return json.dumps(sbom_a, indent=2)
 
 
-if __name__ == "__main__":
-    parser = ArgumentParser()
+def parse_sbom_arg(arg: str, default_flavour: str) -> tuple[str, str]:
+    before_colon, colon, after_colon = arg.partition(":")
+    if colon:
+        flavour = before_colon.lower()
```

Comment:
Do you think it would be helpful if we had a list of valid flavours and validated the inputs according to that list?

Reply:
Although now in tests I also see flavours foo and bar, so we don't want to limit it?

Reply:
In terms of the code, not that much. I think it would allow replacing the `match-case` statement with something like this:

```python
    cachi2_sbom_paths = sbom_paths_by_flavour.get("cachi2", [])
    syft_sbom_paths = sbom_paths_by_flavour.get("syft", [])

    if not syft_sbom_paths or len(cachi2_sbom_paths) > 1:
        flavours = " X ".join(flavour for flavour, _ in sboms)
        raise ValueError(
            f"Unsupported combination of SBOM flavours: {flavours}\n"
            "\n"
            "This script supports merging 0 or 1 cachi2 SBOM with >=1 syft SBOMs"
        )

    if cachi2_sbom_paths:
        merged = merge_syft_and_cachi2_sboms(syft_sbom_paths, cachi2_sbom_paths[0])
    else:
        merged = merge_n_syft_sboms(syft_sbom_paths)
```

But the `match-case` statement seems more obvious